argoverse/utils/sim2.py

```diff
@@ -108,3 +110,25 @@ def transform_from(self, point_cloud: np.ndarray) -> np.ndarray:
     def transform_point_cloud(self, point_cloud: np.ndarray) -> np.ndarray:
         """Alias for `transform_from()`, for synchrony w/ API provided by SE(2) and SE(3) classes."""
         return self.transform_from(point_cloud)
+
+    @classmethod
+    def from_json(cls, json_fpath: Union[str, "os.PathLike[str]"]) -> "Sim2":
+        """Generate class inst. from a JSON file containing Sim(2) parameters as flattened matrices (row-major)."""
+        with open(json_fpath, "r") as f:
+            json_data = json.load(f)
+
+        R = np.array(json_data["R"]).reshape(2, 2)
+        t = np.array(json_data["t"]).reshape(2)
+        s = float(json_data["s"])
+        return cls(R, t, s)
+
+    @classmethod
+    def from_matrix(cls, T: np.ndarray) -> "Sim2":
```

Comment:
How do we handle return typing when a method can raise an exception? I'm not sure what the correct convention is.